sympy/geometry/line.py

```diff
@@ -447,26 +497,25 @@ def perpendicular_segment(self, p):
         >>> p3 in s1
         True
         >>> l1.perpendicular_segment(Point(4, 0))
-        Segment(Point2D(2, 2), Point2D(4, 0))
+        Segment2D(Point2D(2, 2), Point2D(4, 0))
+        >>> from sympy import Point3D, Line3D
+        >>> p1, p2, p3 = Point3D(0, 0, 0), Point3D(1, 1, 1), Point3D(0, 2, 0)
+        >>> l1 = Line3D(p1, p2)
+        >>> s1 = l1.perpendicular_segment(p3)
+        >>> l1.is_perpendicular(s1)
+        True
+        >>> p3 in s1
+        True
+        >>> l1.perpendicular_segment(Point3D(4, 0, 0))
+        Segment3D(Point3D(4/3, 4/3, 4/3), Point3D(4, 0, 0))
 
         """
-        p = Point(p)
+        p = Point.pointify(p, dimension=self.ambient_dimension)
         if p in self:
             return p
-        a, b, c = self.coefficients
-        if a == 0:  # horizontal
-            p2 = Point(p.x, self.p1.y)
-        elif b == 0:  # vertical
-            p2 = Point(self.p1.x, p.y)
-        else:
-            # ax + by + c = 0
-            y = (-c - a*p.x)/b
-            m = self.slope
-            d2 = 1 + m**2
-            H = p.y - y
-            dx = m*H/d2
-            dy = m*dx
-            p2 = (p.x + dx, y + dy)
+        l = self.perpendicular_line(p)
+        p2, = Intersection(Line(self.p1, self.p2), l)
```

Comment:
`p2, = Intersection(Line(self.p1, self.p2), l)` is a sneaky way of unpacking an iterable with one element.  Since `Intersection` is a set, I cannot do `Intersection(Line(self.p1, self.p2), l)[0]`.  Should I add a comment explaining it is an unpacking, or should I replace it with `p2 = list(Intersection(Line(self.p1, self.p2), l))[0]`?